config/src/test/scala/com/typesafe/config/impl/TestUtils.scala

```diff
@@ -157,7 +157,7 @@ abstract trait TestUtils {
         copy
     }
 
-    protected def checkSerializationCompat[T: ClassTag](expectedHex: String, o: T, changedOK: Boolean = false): Unit = {
```

Comment:
I think this particular test doesn't make a lot of sense. Java serialization formats change between versions. Even minor versions. There was a format change in one of the later versions of Java 7 from the Java 7 minor version right before it. Checking that the serialized output matches some specific output doesn't seem like a great test because you'll get different output for varying versions of Java since they only guarantee you'll be able to deserialize output that was generated with that same exact minor version of Java.